adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1275,16 +1246,12 @@ private boolean promptUser(PromptBehavior prompt) {
      * if the redirectUri from the client does not match the valid redirectUri
      * the client app would not jump to the login page
      * redirectUri format %PREFIX://%PACKAGE_NAME/%SIGNATURE
-     * 
-     * @param request
-     * @throws UsageAuthenticationException
-     * @return true if the redirectUri is valid or fail and throw the AuthenticationException
      */
     private boolean verifyBrokerRedirectUri(final AuthenticationRequest request) throws UsageAuthenticationException {   
         final String methodName = ":verifyBrokerRedirectUri";
         final String inputUri = request.getRedirectUri();
         final String actualUri = getRedirectUriForBroker();
-        String errMsg = "";
+        String errMsg;
```

Comment:
The code path guarantees that the errMsg is only initialized, then it should be final. 
The below code is a giant if/else block, making it final will make whoever wants the change the code a second thought.